documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
 
 
 ## Naming - Swagger Checklist ##
 
-&#x25a2; *M1000*: Plural names MUST only be used to represent collections.
+- [ ] **M1000**: Plural names *MUST* only be used to represent collections.
+
+- [ ] **M1001**: Each operation must contain an OperationId of the form "RESOURCE_METHOD" a.k.a "NOUN_VERB", where "RESOURCE" represents the resource or object the operation addresses, and METHOD is a short name for the operation being performed "Get, Start, Delete, Create, List", etc.
+
+- [ ] **M1002**: If the NOUN in the OperationId "NOUN_VERB" matches with the name of the Model definition then the NOUN in the operationId *MUST* use the plural form of NOUN to avoid collision in the namespace.
+
+- [ ] **M1003**: List operations *MUST* be prefixed with "List" and *MUST NOT* be named as "GetXXXX" operations.
+
+- [ ] **M1004**: List operations *MUST* use consistent names and semantics. List operations *MUST* NOT use any other names. Consistent names and semantics are:
+
+  - `List()` - lists all resources under a subscription.
+  - `ListByResourceGroup()` - list all resources in a resource group within a subscription.
+  - `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+
+- [ ] **M1005**: Get operations *MUST* use the METHOD name "Get"
+
+- [ ] **M1006**: PUT operations *MUST* use the METHOD Name "Create"
+
+- [ ] **M1007**: PATCH operations *MUST* use the METHOD Name "Update"
+
+- [ ] **M1008**: POST operations *MUST* use a METHOD Name that matches the semantic operation on the resource, for example "Start" or "Stop"
 
-&#x25a2; *M1001*: List operations MUST be prefixed with "List" and MUST not be named as "GetXXXX" operations.
+- [ ] **M1009**: DELETE operations *MUST* use the METHOD Name "Delete"
 
-&#x25a2; *M1002*: List operations MUST use consistent names and semantics. List operations MUST NOT use any other names. Consistent names and semantics are:
+- [ ] **M1010**: Terminology in Swagger and Azure Portal *MUST* be the same.
 
-- `List()` - lists all resources under a subscription.
-- `ListByResourceGroup()` - list all resources in a resource group within a subscription.
-- `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+- [ ] **M1011**: Resource Model name *MUST* be same as singular form of the resource type. Like for virtualMachines resource type it should be VirtualMachine.
 
-&#x25a2; *M1003*: Terminology in Swagger and Azure Portal MUST be the same.
+- [ ] **M1012**: Parameter names in the apis *MUST* be name of the resource like virtualMachines\{virtualMachine} or virtualMachines\{virtualMachineName} both are good example but virtualMachines\{vm} is bad example. 
 
+- [ ] **M1013**: Any action name *MUST* be same as what is defined in the Rest APIs, like regenerateAdminKeys should be regenerateAdminKeys in the Swagger as well, where as regenerateKeys is bad example.
```

Comment:
Please clarify what an "action" is in this case since it is neither a REST nor a Swagger term to the best of my knowledge. If we are referring to resource providers, again, we need to separate that guidance from general swagger guidance.